tests/phpunit/CRM/Core/FormTest.php

```diff
@@ -11,76 +11,87 @@ class CRM_Core_FormTest extends CiviUnitTestCase {
    * on the screen, but which are hidden when using popup forms.
    * So no assertions required.
    *
-   * @param string $classname
-   * @param callable $additionalSetup
-   *   Function that performs some additional setup steps specific to the form
+   * @param string $url
    *
-   * @dataProvider formClassList
+   * @dataProvider formList
    */
-  public function testOpeningForms(string $classname, callable $additionalSetup) {
-    $form = $this->getFormObject($classname);
+  public function testOpeningForms(string $url) {
+    $this->createLoggedInUser();
 
-    // call the callable parameter we were passed in
-    $additionalSetup($form);
+    $_SERVER['REQUEST_URI'] = $url;
+    $urlParts = explode('?', $url);
+    $_GET['q'] = $urlParts[0];
 
-    // typical quickform/smarty flow
-    $form->preProcess();
-    $form->buildQuickForm();
-    $form->setDefaultValues();
-    $form->assign('action', $form->_action ?? CRM_Core_Action::UPDATE);
-    $form->getTemplate()->fetch($form->getTemplateFileName());
+    $parsed = [];
+    parse_str($urlParts[1], $parsed);
+    foreach ($parsed as $param => $value) {
+      $_REQUEST[$param] = $value;
+    }
+
+    $item = CRM_Core_Invoke::getItem([$_GET['q']]);
+    ob_start();
+    CRM_Core_Invoke::runItem($item);
+    ob_end_clean();
+
+    foreach ($parsed as $param => $dontcare) {
+      unset($_REQUEST[$param]);
+    }
```

Comment:
And I also added this since otherwise these persist across the whole test suite. They don't seem to have broken anything, but this seems cleaner.